### PR TITLE
New color picker for HS and color temp

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

- [x] Add pickers to gallery
- [x] Update more info light color view

### Color light picker demo 

https://github.com/home-assistant/frontend/assets/5878303/b0bdf1fd-d331-43a6-816e-39eb6eccede2

### Color temperature color wheel 
Demo : https://deploy-preview-16549--home-assistant-gallery.netlify.app/#components/ha-temp-color-picker

CleanShot 2023-05-23 at 12 15 53

### Hue Saturation color wheel
Demo : https://deploy-preview-16549--home-assistant-gallery.netlify.app//#components/ha-hs-color-picker

CleanShot 2023-05-23 at 12 15 58

These controls are not accessible, we need to fallback to two sliders for hue saturation and one slider for color temperature in the more-info light. BTW, the current color wheel is not accessible too.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-hs-color-picker.ts [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+  @property({ type: Number, attribute: false })
+  public renderSize?: number;
+
+  @property({ type: Number })
+  public value?: [number, number];
+
+  @property({ type: Number })
+  public colorBrightness?: number;
+
+  @query("#canvas") private _canvas!: HTMLCanvasElement;
```
Comment (bramkragten) [inline src/components/ha-hs-color-picker.ts:70]:
```suggestion
  @query("#canvas", true) private _canvas!: HTMLCanvasElement;
```

**Comment by piitaya** [comment]
> Should `ha-color-picker` be removed?

Yes good point

**Comment by bramkragten** [comment]
Should the `hs` and `ct` pickers have a common super class that they extend, as there is a lot of overlap, it could make maintaining easier? Or would that only make it more complicated?

File: src/components/ha-hs-color-picker.ts (hunk 1 above)
Comment (piitaya) [reply src/components/ha-hs-color-picker.ts:70]: It doesn't work with cache 😕

**Comment by bramkragten** [comment]
> > Should the `hs` and `ct` pickers have a common super class that they extend, as there is a lot of overlap, it could make maintaining easier? Or would that only make it more complicated?
> 
> I think we can do this. I preferred to duplicate for now for easy development but once it's stable we can consider it. In this PR or another one ?

Let's do it in another one

**Review by bramkragten — APPROVED** [review]

**Comment by W-Floyd** [comment]
Why is the color temperature picker a wheel? As a single dimensional value, shouldn't it be a slider? It makes it difficult to exactly set the maximum/minimum color temperature.

**Pull request merged**